src/components/project/project.service.ts

```diff
@@ -578,16 +605,26 @@ export class ProjectService {
     // TODO: re-connect the locationId node when locations are hooked up
 
     if (input.primaryLocationId) {
-      const location = await this.locationService.readOne(
-        input.primaryLocationId,
-        session
-      );
-
-      if (!location.fundingAccount.value)
-        throw new InputException(
-          'Cannot connect location without a funding account',
-          'project.primaryLocationId'
+      try {
+        const location = await this.locationService.readOne(
+          input.primaryLocationId,
+          session
         );
+        if (!location.fundingAccount.value) {
+          throw new InputException(
+            'Cannot connect location without a funding account',
+            'project.primaryLocationId'
+          );
+        }
+      } catch (e) {
+        if (e instanceof NotFoundException) {
+          throw new NotFoundException(
+            'Primary location not found',
+            'project.primaryLocationId'
```

Comment:
Let's forward the exception we are wrapping on, so we can see the true error source in the logs
```suggestion
            'project.primaryLocationId',
            e
```